alexk_low_cost_robot/wizard/src/main.rs

```diff
@@ -0,0 +1,158 @@
+    } else {
+        // x430 for 2 first and xl330 for the rest
+
+        let mut pos = xl430::sync_read_present_position(
+            &io,
+            serial_port,
+            &[0, 1],
+        ).expect("Read Communication error");
+
+        let pos2 = xl330::sync_read_present_position(
+            &io,
+            serial_port,
+            &[2, 3, 4, 5],
+        ).expect("Read Communication error");
+
+        pos.iter().chain(pos2.iter()).map(|&x| convert_to_extended_mode(x)).collect::<Vec<_>>()
+    };
+
+    pos
+}
+
+fn prepare_configuration(io: &DynamixelSerialIO, serial_port: &mut dyn SerialPort, puppet: bool) {
+    xl330::sync_write_torque_enable(
+        &io,
+        serial_port,
+        &[2, 3, 4, 5],
+        &[0; 4],
+    ).expect("Communication error");
+
+    // Now we enable all modes to "position extended"
+
+    xl330::sync_write_operating_mode(
+        &io,
+        serial_port,
+        &[2, 3, 4, 5],
+        &[4; 4],
+    ).expect("Communication error");
+
+    if !puppet {
+        xl330::sync_write_torque_enable(
+            &io,
+            serial_port,
+            &[0, 1],
+            &[0; 2],
+        ).expect("Communication error");
+
+        xl330::sync_write_operating_mode(
+            &io,
+            serial_port,
+            &[0, 1],
+            &[4; 2],
+        ).expect("Communication error");
+    } else {
+        xl430::sync_write_torque_enable(
+            &io,
+            serial_port,
+            &[0, 1],
+            &[0; 2],
+        ).expect("Communication error");
+
+        xl430::sync_write_operating_mode(
+            &io,
+            serial_port,
+            &[0, 1],
+            &[4; 2],
+        ).expect("Communication error");
+    }
+}
+
+fn get_nearest_rounded_positions (pos: &Vec<i32>) -> Vec<i32> {
+    pos.iter().map(|&x| {
+        if x >= 0 {
+            let k = x / 1024;
+            if x % 1024 > 512 {
+                (k + 1) * 1024
+            } else {
+                k * 1024
+            }
+        } else {
+            let k = (-x) / 1024;
+            if (-x) % 1024 > 512 {
+                -(k + 1) * 1024
+            } else {
+                -k * 1024
+            }
+        }
+    }).collect::<Vec<_>>()
+}
+
+fn invert(pos: &Vec<i32>, inverted: &[bool; 6]) -> Vec<i32> {
+    pos.iter().enumerate().map(|(i, &x)| {
+        if inverted[i] {
+            -x
+        } else {
+            x
+        }
+    }).collect::<Vec<_>>()
+}
+
+fn configure_homing(io: &DynamixelSerialIO, serial_port: &mut dyn SerialPort, inverted: &[bool; 6], puppet: bool) {
+    println!("------Configuring homing");
+
+    prepare_configuration(io, serial_port, puppet);
+}
+
+fn main() {
+    let cli = Cli::parse();
+    println!("CLI is {:#?}", cli);
+
+    // Configure the serial port
+    let mut serial_port = serialport::new(cli.port, 1_000_000).timeout(Duration::from_millis(200)).open().expect("Failed to open port");
+    let io = DynamixelSerialIO::v2();
+}
```

Comment:
I think that there is a formatting issue at the end with missing end of line at the end of the file.

Are you using rustfmt? formatting on save?

Reply:
My bad, I use the Jetbrain formatter, I thought it was the same as rustfmt, but i'll change for this one